Dockerfile

```diff
@@ -1,12 +1,36 @@
-# Tomcat image
-FROM tomcat:8-jre8-temurin-focal
+# STAGE 0
+FROM rockylinux:9 as base
 
-COPY target/cfe_18.war ${CATALINA_HOME}/webapps/
+# Rocky linux 9 does not come with java at all. Wget for downloading package
+RUN dnf update -y && dnf install wget java-11-openjdk-headless -y && dnf clean all
+# Tomcat user
+RUN useradd -r -m -U -d /opt/tomcat -s /bin/false tomcat
 
-EXPOSE 8080
+ENV TOMCAT_MAJOR 9
+ENV TOMCAT_VERSION 9.0.98
+ENV CATALINA_HOME /opt/tomcat/apache-tomcat-${TOMCAT_VERSION}/
+RUN mkdir -p "${CATALINA_HOME}"
+WORKDIR ${CATALINA_HOME}
+
+# WGET and verify
+RUN wget -cq https://downloads.apache.org/tomcat/tomcat-${TOMCAT_MAJOR}/v${TOMCAT_VERSION}/KEYS
+RUN gpg2 --import KEYS
+RUN wget -cq https://downloads.apache.org/tomcat/tomcat-${TOMCAT_MAJOR}/v${TOMCAT_VERSION}/bin/apache-tomcat-${TOMCAT_VERSION}.tar.gz
+RUN wget -cq https://downloads.apache.org/tomcat/tomcat-${TOMCAT_MAJOR}/v${TOMCAT_VERSION}/bin/apache-tomcat-${TOMCAT_VERSION}.tar.gz.asc
```

Comment:
The downloads.apache.org link will break eventually as new versions are released so use https://archive.apache.org/dist/tomcat/tomcat-9/ as a base instead